packages/jest-runtime/src/should_instrument.js

```diff
@@ -35,7 +35,10 @@ export default function shouldInstrument(
     return true;
   }
 
-  if (config.testRegex && filename.match(config.testRegex)) {
+  if (
+    config.testRegex &&
+    config.testRegex.some(regex => filename.match(regex))
```

Comment:
I know you didn't introduce it, but can you use `test` instead?

Reply:
To do this efficiently, it was necessary to change the internal type in `ProjectConfig` to an array of `RegExp` instead of `string` (rather than parse the strings on every use here). This makes more sense anyway, but it touched a few things.